### ISSUE TITLE
SemanticSim: under runTest the default RealDriverClock disagrees with the test scheduler and flips handshake outcomes (4/6 seeds)

### ISSUE THREAD

**Issue by thebehera** [DitchOoM/socket#497]
Split out of #480's "second finding" so it is tracked on its own; the #480 fix (a stalled handshake past the caller's bound closes typed) deliberately does not touch it.

## The measurement

`HandshakeIdleTimeoutSimTests.realAndVirtualClocksAreComparedRatherThanAssumedEquivalent` runs the same six seeds through `withSemanticSim` (real quiche client + real quiche server over `ImpairedPipe`, `loss = 0.30`, `latency = 5ms`, `idleTimeout = 2s`, 60s establishment bound) twice, varying only the driver clock:

```
[#450 clock comparison] loss=0.30, 6 seeds
  seed 1: virtual=Established  real=Threw(kind=TimeoutCancellationException)
  seed 2: virtual=Established  real=Threw(kind=TimeoutCancellationException)
  seed 3: virtual=Established  real=Established
  seed 4: virtual=LocalIdleTimeout  real=Threw(kind=TimeoutCancellationException)
  seed 5: virtual=Established  real=Established
  seed 6: virtual=Established  real=Threw(kind=TimeoutCancellationException)
  disagreements: 4 [1, 2, 4, 6]
```

(Measured 2026-08-28 on `origin/main` = 91588acf; the issue body of #480 recorded 3/6 on an earlier run. The test records the measurement and deliberately does not assert the two columns equal.)

## Mechanism, as currently understood

`withSemanticSim` defaults to `RealDriverClock`. Under `runTest`, everything on our side of the FFI — the pipe's impairment `delay()`s, the driver's `select { onTimeout }` wakes, the establishment `withTimeout` — runs on the test scheduler's **virtual** time and fast-forwards, while quiche's internal timers (PTO, loss detection, idle) read a **wall** clock that has barely moved. The two sides of the FFI disagree about how much time has elapsed, so:

- the establishment bound (virtual) can expire before quiche (wall) has even reached its first PTO retransmit, which is what the `TimeoutCancellationException` column is; and
- seed 4, which under the virtual clock produces the #450 signature (`Closed(ByLocal(IdleTimeout))` deterministically, 5/5), is indistinguishable from the other three real-clock timeouts.

`SimClock` (which also feeds `CallerClockQuicheApi`, so quiche's own `Instant::now()` reads the scheduler) makes the two sides agree, and that is why the `#450` seed instrument pins `clock = SimClock(testScheduler)` explicitly.

## Why it matters beyond that one test

Every existing `withSemanticSim` result taken under the default `RealDriverClock` under `runTest` deserves a re-read with this in mind: a result that depends on any quiche timer (PTO, idle, path validation) was taken with the two clocks incoherent. The `SemanticSim` KDoc already calls the real-clock default "a wiring choice, not a limitation".

## Possible directions (not decided here)

- Make `withSemanticSim`'s clock **non-defaulted** so every sim test states which clock it ran on.
- Or default it to `SimClock` whenever it is invoked under a `TestScope`, keeping `RealDriverClock` only for callers on real dispatchers.
- Either way, the comparison test above is the regression instrument: the `disagreements` count is the thing to watch.

Related: #480 (the finding's origin), #450 (the seed instrument), the `SemanticSim.kt` virtual-time status KDoc.
